docs/plugin/errors.md

```diff
@@ -0,0 +1,506 @@
+| -32000 to -32099 | Server error     | Reserved for implementation-defined server-errors.           |
+
+In the below example, an attempt is made in activate a non-existent plugin. The Controller plugin returns `ERROR_UNKNOWN_KEY`
+
+:arrow_right: Request
+
+```json
+{
+	"jsonrpc": "2.0",
+	"id": 1,
+	"method": "Controller.1.activate",
+	"params": {
+		"callsign": "fakePlugin"
+	}
+}
+```
+
+:arrow_left: Response
+
+```json
+{
+	"jsonrpc": "2.0",
+	"id": 1,
+	"error": {
+		"code": 22,
+		"message": "ERROR_UNKNOWN_KEY"
+	}
+}
+```
+
+## Handling Unexpected COM-RPC Disconnections
+
+From the perspective of a plugin, there are two COM-RPC disconnection scenarios to consider:
+
+* When the out-of-process side of the plugin unexpectedly terminates (either due to a crash or being killed by an external entity such as the linux OOM killer)
+* When a client that has registered for notifications crashes unepxectedly.
+
+Whilst the framework can detect unexpected COM-RPC disconnects and handle updating reference counts accordingly, there are actions that should be taken in the plugin code to ensure safety.
+
+### Out-Of-Process Disconnects
+
+!!! warning
+	It is essential that plugins implement this feature if they are expected to run out-of-process
+
+If your plugin implements an interface that could run out of process, then it is very important the plugin subscribes to the remote connection notification `RPC::IRemoteConnection::INotification `. This is a common pattern that you will see across many plugins.
+
+The framework will raise this notification whenever a COM-RPC connection or disconnection occurs. The plugin should check if the disconnected connection belongs to them, and if so take action. Typically, this action will be for the plugin to deactivate itself with a `Failure` reason. 
+
+Without listening for this notification and taking action, if the out-of-process side of a plugin dies the plugin itself will not be deactivated.
+
+By deactivating with the `Failure` reason, the post-mortem handler will kick in. This will dump the contents of some system files to the log (memory information, load averages) to help debugging.
+
+!!! note
+	The `Deactivated()` function might be called on a socket thread which we do not want to block. As a result, this function should return as quickly as possible and any work that needs doing (e.g. deactivating the plugin) must be done on a separate thread in the main worker pool.
+
+Example:
+
+```c++ linenums="1" title="TestPlugin.h" hl_lines="3-30 60"
+class TestPlugin : public PluginHost::IPlugin, public PluginHost::JSONRPC {
+private:
+    class Notification : public RPC::IRemoteConnection::INotification {
+    public:
+        Notification() = delete;
+        explicit Notification(TestPlugin* parent)
+            : _parent(*parent)
+        {
+            ASSERT(parent != nullptr);
+        }
+
+        ~Notification() = default;
+        Notification(const Notification&) = delete;
```

Comment:
We certainly would also delete the assignment operator and nowadays we also delete the move operator and assignment operator (technically not needed but to make it more clear that if you would implement one or the other then it would be needed (but that last part we certainly have not documented and you would only see in very new code)